internal/services/v1/grouping.go

```diff
@@ -26,50 +25,51 @@ type groupingParameters struct {
 
 // groupItems takes a slice of BulkCheckPermissionRequestItem and groups them based
 // on using the same permission, subject type, subject id, and caveat.
-func groupItems(ctx context.Context, params groupingParameters, items []*v1.BulkCheckPermissionRequestItem) ([]groupedCheckParameters, error) {
-	groups := mapz.NewMultiMap[string, string]()
-	groupParams := make(map[string]computed.CheckParameters, len(items))
+func groupItems(ctx context.Context, params groupingParameters, items []*v1.BulkCheckPermissionRequestItem) (map[string]*groupedCheckParameters, error) {
```

Comment:
I did change the signature here to return a map of `hash` -> `groupedCheckParameters`. Otherwise I would need to convert the dictionary back to a slice in the end with `maps.Values` which I think would cancel out the memory savings.